tools/hack/get-egctl.sh

```diff
@@ -0,0 +1,208 @@
+      TAG=$(curl -Ls $latest_release_url | grep 'href="/envoyproxy/gateway/releases/tag/v[0-9]*.[0-9]*.[0-9]*\"' | sed -E 's/.*\/envoyproxy\/gateway\/releases\/tag\/(v[0-9\.]+)".*/\1/g' | head -1)
+    elif [ "${HAS_WGET}" == "true" ]; then
+      TAG=$(wget $latest_release_url -O - 2>&1 | grep 'href="/envoyproxy/gateway/releases/tag/v[0-9]*.[0-9]*.[0-9]*\"' | sed -E 's/.*\/envoyproxy\/gateway\/releases\/tag\/(v[0-9\.]+)".*/\1/g' | head -1)
+    fi
+  else
+    TAG=$DESIRED_VERSION
+  fi
+}
+
+# checkEGCTLInstalledVersion checks which version of egctl is installed and
+# if it needs to be changed.
+checkEGCTLInstalledVersion() {
+  if [[ -f "${EGCTL_INSTALL_DIR}/${BINARY_NAME}" ]]; then
+    version=$("${EGCTL_INSTALL_DIR}/${BINARY_NAME}" version | grep -Eo "v[0-9]+\.[0-9]+.*" )
+    if [[ "$version" == "$TAG" ]]; then
+      echo "egctl ${version} is already ${DESIRED_VERSION:-latest}"
+      return 0
+    else
+      echo "egctl ${TAG} is available. Changing from version ${version}."
+      return 1
+    fi
+  else
+    return 1
+  fi
+}
+
+# downloadFile downloads the latest binary package
+# for that binary.
+downloadFile() {
+  EGCTL_DIST="egctl_${TAG}_${OS}_${ARCH}.tar.gz"
+  DOWNLOAD_URL="https://github.com/envoyproxy/gateway/releases/download/$TAG/$EGCTL_DIST"
+  EGCTL_TMP_ROOT="$(mktemp -dt egctl-installer-XXXXXX)"
+  EGCTL_TMP_FILE="$EGCTL_TMP_ROOT/$EGCTL_DIST"
+  echo "Downloading $DOWNLOAD_URL"
+  if [ "${HAS_CURL}" == "true" ]; then
+    curl -SsL "$DOWNLOAD_URL" -o "$EGCTL_TMP_FILE"
+  elif [ "${HAS_WGET}" == "true" ]; then
+    wget -q -O "$EGCTL_TMP_FILE" "$DOWNLOAD_URL"
+  fi
+}
+
+# installFile installs the egctl binary.
+installFile() {
+  EGCTL_TMP="$EGCTL_TMP_ROOT/$BINARY_NAME"
+  mkdir -p "$EGCTL_TMP"
+  tar xf "$EGCTL_TMP_FILE" -C "$EGCTL_TMP"
+  EGCTL_TMP_BIN="$EGCTL_TMP/bin/$OS/$ARCH/egctl"
+  echo "Preparing to install $BINARY_NAME into ${EGCTL_INSTALL_DIR}"
+  runAsRoot cp "$EGCTL_TMP_BIN" "$EGCTL_INSTALL_DIR/$BINARY_NAME"
+  echo "$BINARY_NAME installed into $EGCTL_INSTALL_DIR/$BINARY_NAME"
+}
+
+# fail_trap is executed if an error occurs.
+fail_trap() {
+  result=$?
+  if [ "$result" != "0" ]; then
+    if [[ -n "$INPUT_ARGUMENTS" ]]; then
+      echo "Failed to install $BINARY_NAME with the arguments provided: $INPUT_ARGUMENTS"
+      help
+    else
+      echo "Failed to install $BINARY_NAME"
+    fi
+    echo -e "\tFor support, go to https://github.com/envoyproxy/gateway."
+  fi
+  cleanup
+  exit $result
+}
+
+# testVersion tests the installed client to make sure it is working.
+testVersion() {
+  set +e
+  if ! [ "$(command -v $BINARY_NAME)" ]; then
+    echo "$BINARY_NAME not found. Is $EGCTL_INSTALL_DIR on your PATH?"
+    exit 1
+  fi
+  set -e
+}
+
+# help provides possible cli installation arguments
+help () {
+  echo "Accepted cli arguments are:"
+  echo -e "\t[--help|-h ] ->> prints this help"
+  echo -e "\t[--version|-v <desired_version>] . When not defined it fetches the latest release from GitHub"
```

Comment:
thoughts on making VERSION also an env var ?
e.g. how Istio does it
```
curl -L https://istio.io/downloadIstio | ISTIO_VERSION=1.17.1 TARGET_ARCH=x86_64 sh -
```

Reply:
Sounds good. Done.